scripts/core/ui/components/generic-form/input-types/select_single_value_autocomplete.tsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import classNames from 'classnames';
+import {IInputType} from '../interfaces/input-types';
+import {AutoComplete} from '../../Autocomplete';
+import {IRestApiResponse} from 'superdesk-api';
+
+type IProps = IInputType<string>;
+
+interface IState {
+    item?: any;
+}
+
+export function getSelectSingleValueAutoComplete(
+    query: (searchString: string, props: IProps) => Promise<IRestApiResponse<any>>,
+    queryById: (id: string) => Promise<any>,
+    getPlaceholder: (props: IProps) => string,
+    getLabel: (item) => string,
+    getDependentFields?: (props: IProps) => Array<string>,
+) {
```

Comment:
just wondering why having this factory instead of those being props of a component?

Reply:
because I need the component to conform to a [common interface](https://github.com/superdesk/superdesk-client-core/blob/develop/scripts/core/ui/components/generic-form/form-field.tsx#L17)